core-active-state.html

```diff
@@ -0,0 +1,149 @@
+       *
+       * @attribute active
+       * @type boolean
+       * @default false
+       */
+      active: {
+        type: Boolean,
+        notify: true,
+        readOnly: true,
+        observer: 'activeChanged',
+        value: false
+      },
+
+      /**
+       * If true, the user is currently holding down the button.
+       *
+       * @attribute pressed
+       * @type boolean
+       * @default false
+       */
+      pressed: {
+        type: Boolean,
+        value: function() {
+          this._setPressed(this.pressed || false);
+        },
+        readOnly: true,
+        observer: 'pressedChanged'
+      },
+
+      /**
+       * If true, the button toggles the active state with each tap.
+       * Otherwise, the button becomes active when the user is holding
+       * it down.
+       *
+       * @attribute toggle
+       * @type boolean
+       * @default false
+       */
+      toggle: {
+        type: Boolean,
+        value: false
+      },
+
+      /**
+       * If true, the user cannot interact with this element.
+       *
+       * @attribute disabled
+       * @type boolean
+       * @default false
+       */
+      disabled: {
+        type: Boolean,
+        observer: 'disabledChanged',
+        value: false
+      },
+
+      boundDownHandler: {
+        value: function() {
+          return this.onTargetDown.bind(this);
+        }
+      },
+
+      boundUpHandler: {
+        value: function() {
+          return this.onTargetUp.bind(this);
+        }
+      }
+    },
+
+    get target() {
+      return this.host || this.parentNode;
+    },
+
+    ready: function() {
+      this.afterReady = true;
+      // TODO(cdata): We this as a workaround for polymer/polymer#1285
+      if (this.isAttached) {
+        this.attached();
+      }
+    },
+
+    attached: function() {
+      if (!this.afterReady) {
+        return;
+      }
+
+      this.target.addEventListener('mousedown', this.boundDownHandler, this);
+      this.target.addEventListener('mouseup', this.boundUpHandler, this);
+    },
+
+    detached: function() {
+      this.target.removeEventListener('mousedown', this.boundDownHandler, this);
+      this.target.removeEventListener('mouseup', this.boundUpHandler, this);
+    },
+
+    onTargetDown: function() {
+      this._setPressed(true);
+    },
+
+    onTargetUp: function() {
+      this._setPressed(false);
+    },
+
+    pressedChanged: function() {
+      if (this.toggle) {
+        if (this.pressed) {
+          this._setActive(!this.active);
+        }
+      } else {
+        this._setActive(this.pressed);
+      }
+    },
+
+    activeChanged: function() {
+      if (this.active) {
+        this.target.setAttribute('active', '');
+      } else {
+        this.target.removeAttribute('active');
+      }
+    },
+
+    disabledChanged: function() {
+      if (this.disabled) {
+        this._setPressed(false);
+        this._setActive(false);
+      }
+    },
```

Comment:
Yeah, sorry. I have to do a lot of mental swapping back and forth because @nevir and @justinfagnani prefer to have the trailing commas. It's a mad house.